### PR TITLE
Add visibility editor to card editor

### PR DESCRIPTION
**Pull request by piitaya**
## Proposed change

Add visibility editor to every card editor.
The visibility tab is not visible for conditional card to avoid confusion.
It's also not visible for nested card (inside conditional card, stack card)

Also, the responsiveness on the dialog has been improved with bigger dialog on desktop.

CleanShot 2024-05-30 at 16 45 10
CleanShot 2024-05-30 at 16 45 34

Not available for nested cards

CleanShot 2024-05-30 at 16 45 57

Not available for conditional card

CleanShot 2024-05-30 at 16 39 39

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml

```

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue or discussion:
- Link to documentation pull request:

## Checklist

- [ ] The code change is tested and works locally.
- [ ] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

## Summary by CodeRabbit

- **New Features**
  - Introduced tab handling for configuration and visibility in the card editor.
  - Added a new visibility editor for Lovelace cards to manage visibility conditions.

- **Enhancements**
  - Updated rendering logic to support the new tabbed interface for better user experience.
  - Improved translations for configuration and visibility tabs.

- **Bug Fixes**
  - Adjusted CSS properties for dialogs and content to improve display on different screen sizes.

### CODE REVIEW

File: src/panels/lovelace/editor/card-editor/hui-card-element-editor.ts [hunk 1]
```diff
@@ -1,11 +1,22 @@
...
+import "./hui-card-visibility-editor";
+
+const TABS = ["config", "visibility"] as const;
 
 @customElement("hui-card-element-editor")
 export class HuiCardElementEditor extends HuiElementEditor<LovelaceCardConfig> {
+  @state() private _curTab: (typeof TABS)[number] = TABS[0];
+
+  @property({ type: Boolean, attribute: "show-visibility-tab" })
+  public showVisibilityTab: boolean = false;
```
[the remark this answers is not in the document]
Comment (piitaya) [reply src/panels/lovelace/editor/card-editor/hui-card-element-editor.ts:18]: Ok good idea ! Can you make a suggestion?

**Comment by noxhirsch** [comment]
I have to say, it would be very helpful for me to be able to set the visibility in nested cards without conditional cards. Among other things, I have a large vertical stack of cards on my wall tablets, which are visible depending on the room and other conditions. If the tabs don't fit there, I could also imagine an eye icon next to the YAML editor icon for switching.

File: src/panels/lovelace/editor/card-editor/hui-card-element-editor.ts [hunk 2]
```diff
@@ -27,6 +38,73 @@ export class HuiCardElementEditor extends HuiElementEditor<LovelaceCardConfig> {
...
+          <hui-card-visibility-editor
+            .hass=${this.hass}
+            .config=${this.value}
+            @value-changed=${this._configChanged}
+          ></hui-card-visibility-editor>
+        `;
+        break;
+    }
+    return html`
+      <paper-tabs
```
Comment (bramkragten) [inline src/panels/lovelace/editor/card-editor/hui-card-element-editor.ts:74]: Can we use `mwc-tabs` instead?

File: src/panels/lovelace/editor/hui-element-editor.ts [hunk 3]
```diff
@@ -198,6 +199,10 @@ export abstract class HuiElementEditor<T, C = any> extends LitElement {
     return this.value ? (this.value as any).type : undefined;
   }
 
+  protected renderConfigElement(): TemplateResult {
+    return html`${this._configElement}`;
```
Comment (bramkragten) [inline src/panels/lovelace/editor/hui-element-editor.ts:203]:
```suggestion
    return this._configElement;
```

**Review by bramkragten — APPROVED** [review]

**Pull request merged**